base_rest/controllers/main.py

```diff
@@ -138,8 +138,8 @@ def _validate_method_name(self, method_name):
             raise BadRequest()
         return True
 
-    def _process_method(self, service_name, method_name, *args, params=None):
+    def _process_method(self, service_name, method_name, *args, params=None, **kwargs):
```

Comment:
This signature change will break a lot of projets....

Reply:
@etobella `params_as_kwargs' should not be an option since we have a real issue here. That's why I try to find a solution to solve the issue without introducing a new parameter into the route definition nor modifying the methods signatures.

Reply:
Solution is not simple, probably not possible on current installations.

With the current app, you could do the following:

```

    @restapi.method(
        [(["/get/<int:id", "GET")],
    )
    def get_data(self, _id):
```

and worked properly, you can see that the parameters have not the same name, as it depended on order, my proposal would be to set the params_as_kwargs as False and fix the issue on 15.0

Reply:
For example, even on the tests of rest-framework, exists a case when the parameter name has been changed:

https://github.com/OCA/rest-framework/blob/12.0/base_rest/tests/test_openapi_generator.py#L32

Also, the automatic generated calls, like `get` have the same issue as we can see here:
https://github.com/OCA/rest-framework/blob/13.0/base_rest/models/rest_service_registration.py#L236
https://github.com/shopinvader/odoo-shopinvader/blob/14.0/shopinvader/services/address.py#L31
We are expecting the field `_id` but on the url we are expecting `id`

My point is, we cannot use a dictionary as it can break current installations (the name used might be different) but  we should use key word arguments. Once we have arrived to this contradiction, there is not many options.

IMO, my proposal is similar to queue_job new implementation, when the old behavior had to be maintained, but the new logic will be the only one present after 15.0.

Another option is to show a message when it not fulfill the new behavior, this way, people have time to solve the issue on their installations

Reply:
Thank you @etobella for all this hard work. I'll take time to dig into this problem on mys side too. (Just to convince me that there is no other solution :smirk).

> Also, the automatic generated calls, like get have the same issue as we can see here:
> https://github.com/OCA/rest-framework/blob/13.0/base_rest/models/rest_service_registration.py#L236
> https://github.com/shopinvader/odoo-shopinvader/blob/14.0/shopinvader/services/address.py#L31

The generated routes for instance methods could be fixed to declare the path parameter as `_id` since it's the one  expected by the old api.

Reply:
I know we can change them, I was trying to show, that this problem could be generic :sob: